sdk/core/azure-core/inc/logging/logging.hpp

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include "azure.hpp"
+
+#include <algorithm>
+#include <functional>
+#include <set>
+#include <string>
+
+namespace Azure { namespace Core { namespace Logging {
+  class LogClassification;
+
+  typedef std::function<
+      void(LogClassification const& classification, std::string const& message) noexcept>
+      LogListener;
+
+  void SetLogListener(LogListener logListener);
+  void ResetLogListener();
+
+  void SetLogClassifications(std::set<LogClassification> logClassifications);
+  void ResetLogClassifications();
+
+  namespace Details {
+    enum class Facility : uint16_t
+    {
+      Core = 1,
+      Storage = 100,
+    };
+
+    template <Facility> class LogClassifications;
+  } // namespace Details
+
+  class LogClassification {
+    template <Details::Facility> friend class Details::LogClassifications;
+
+    int32_t m_value;
+
+    constexpr explicit LogClassification(Details::Facility facility, int16_t number)
+        : m_value((static_cast<int32_t>(number) << 16) | static_cast<int32_t>(facility))
+    {
+    }
+
+  public:
+    constexpr bool operator<(LogClassification const& other) const
```

Comment:
Is < actually used?
Can we get rid of this operator?

Reply:
If there's no way to order them, they can't be inserted into unique sets such as `std::set` and `std::map`, and also you can't sort them.
We can try to make this more elegant by making `friend struct std::less<LogClassification>` - and moving `operator<` to private. But that won't work for `std::sort` on GCC. But maybe that's ok. Let me try that.

Reply:
I moved `operator<` to private. It is still needed to be able to insert instances of `LogClassification`to `std::set`.

Reply:
Moved `operator<` to private.